app/src/main/java/dracula/vlad/androcalc/MainActivity.java

```diff
@@ -113,106 +100,121 @@ public void afterTextChanged(Editable c) {
+
+        for (final Button button : buttons) {
+            button.setOnClickListener(v -> {
+                int lengthET = inputView.getText().length();
+                String curBtn = button.getText().toString();
+
+                if (!button.getText().toString().equals(EQUATION) && lengthET != 0) {
+                    String lastChar = inputView.getText().toString();
+                    int index = inputView.getSelectionStart();
+                    lastChar = index != 0 ? String.valueOf(lastChar.charAt(index - 1)) : " ";
+
+                    if (!(lastChar.equals(curBtn))) {
+
+                        if (lastChar.equals(DIVISION) && curBtn.equals(MULTIPLICATION)) {
+                            replaceOperator(curBtn);
+                        } else if (lastChar.equals(MULTIPLICATION) && curBtn.equals(DIVISION)) {
+                            replaceOperator(curBtn);
+                        } else if (lastChar.equals(ADDITION) && curBtn.equals(SUBTRACTION)) {
+                            replaceOperator(curBtn);
+                        } else if (lastChar.equals(SUBTRACTION) && curBtn.equals(ADDITION)) {
+                            replaceOperator(curBtn);
+                        } else {
                             addAtCursorPos(curBtn);
                         }
                     }
-                }
-                else if (btn.getText().toString().equals("-")){
+                } else if (button.getText().toString().equals(SUBTRACTION)) {
                     addAtCursorPos(curBtn);
+                } else if (button.getText().toString().equals(EQUATION)) {
+                    inputView.setText(resultView.getText());
+                    userInput = "";
+                    inputView.setSelection(inputView.getText().length());
                 }
-                else if (btn.getText().toString().equals("=")) {
-                    editText.setText(tv_Res.getText());
-                    input="";
-                    editText.setSelection(editText.getText().length());
+            });
+        }
+
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    private void hideSoftwareKeyboard() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            inputView.setShowSoftInputOnFocus(false);
+        } else {
+            inputView.setOnTouchListener((v, event) -> {
+                v.onTouchEvent(event);
+                InputMethodManager inputMethod = (InputMethodManager) v.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+
+                if (inputMethod != null) {
+                    inputMethod.hideSoftInputFromWindow(v.getWindowToken(), 0);
                 }
+
+                return true;
             });
         }
+    }
 
+    private void replaceOperator(String currentInputButton) {
+        delAtCursorPos();
+        addAtCursorPos(currentInputButton);
     }
 
     public void getChildViews(ViewGroup viewGroup) {
```

Comment:
method get means that it is getter for some field. please re-think about usage of this method and give clean name